test/e2e/dra/dra.go

```diff
@@ -51,30 +52,34 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 	ginkgo.Context("kubelet", func() {
 		var driver = NewDriver(f, 1, 1 /* nodes */) // All tests get their own driver instance.
 
-		ginkgo.Context("", func() {
+		ginkgo.When(
+			"external claim is used in the 'wait for first consumer' allocation mode",
+			genExternalClaimTest(ctx, f, driver, corev1.AllocationModeWaitForFirstConsumer))
+
+		ginkgo.When(
+			"external claim is used in the 'immediate' allocation mode",
+			genExternalClaimTest(ctx, f, driver, corev1.AllocationModeImmediate))
+
+		ginkgo.When("internal claim is used", func() {
 			var b = newBuilder(f, driver)
 
 			ginkgo.It("registers plugin", func() {
 				// If we got here, the driver is running.
 			})
 
-			ginkgo.It("supports simple pod", func() {
+			ginkgo.It("supports simple pod referencing inline resource claim", func() {
 				parameters := b.resourceClaimParameters()
 				pod := b.podInline()
 				b.create(ctx, parameters, pod)
-
-				err := e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, pod.Name, pod.Namespace)
-				framework.ExpectNoError(err, "start pod with inline resource claim")
-
-				log, err := e2epod.GetPodLogs(f.ClientSet, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
-				var envStr string
-				for key, value := range b.resourceClaimParametersEnv() {
-					envStr = fmt.Sprintf("\nuser_%s=%s\n", key, value)
-					break
-				}
-				gomega.Expect(log).To(gomega.ContainSubstring(envStr), "container env variables")
+				b.testPod(f.ClientSet, pod)
 			})
 
+			ginkgo.It("supports inline claim referenced by multiple containers", func() {
```

Comment:
Inline claims can also use immediate allocation. It doesn't make sense, but should also get tested.